src/custom-select/index.ts

```diff
@@ -17,41 +17,39 @@ class CustomSelect {
         this.$select.onSelect(event.item.i);
       }
       this.update({ isActive: false });
-      this.refs.toggle.blur();
     }
   };
   isActive: boolean = false;
 
+  init() {
+    utils.WINDOW.addEventListener('click', this.onClickDeactivate);
+  }
+
   onHoverActivate(event: MouseEvent & Tag.Event) {
     event.preventUpdate = true;
     if (this.props.hover && !this.isActive) {
       this.update({ isActive: true });
     }
   }
 
-  onClickDeactivate(event: MouseEvent & Tag.Event) {
+  onHoverDeactivate(event: MouseEvent & Tag.Event) {
     event.preventUpdate = true;
     if (!this.props.hover && !this.isActive) {
       this.update();
     }
   }
 
-  onLostFocus = (event: FocusEvent & Tag.Event) => {
+  onClickDeactivate = (event: MouseEvent & Tag.Event) => {
     event.preventUpdate = true;
-    if (!this.props.hover) {
-      this.isActive = false;
+    if (!this.props.hover && event.target !== this.refs.toggle.querySelector('button')) {
```

Comment:
we can probably make it easier to access this button in a way that doesn't require a new `querySelector` every time. If you look at the `gb-button` template you'll notice it actually already has a reference to its own inner `<button>` element.

Thus, we can even remove our own `ref="toggle"` from the template, remove the `refs: { toggle: HTMLButtonElement };` property typing and change the line here to use `this.tags['gb-button'].refs.button`